tests/unit/mainwindow/test_prompt.py

```diff
@@ -94,6 +97,31 @@ def test_backspacing_path(self, qtbot, tmp_path, get_prompt):
         prompt.item_focus('next')
         assert prompt._lineedit.text() == str(testdir / 'bar')
 
+    @pytest.mark.parametrize("keys,expected",
+                            [([], ['bar', 'bat', 'foo']),
+                            ([Qt.Key_F], ['foo']),
+                            ([Qt.Key_A], ['bar', 'bat'])])
+    def test_filtering_path(self, qtbot, tmp_path, get_prompt, keys, expected):
+        testdir = tmp_path / 'test'
+
+        for directory in ['bar', 'foo', 'bat']:
+            (testdir / directory).mkdir(parents=True)
+
+        prompt = get_prompt(str(testdir) + os.sep)
+        for key in keys:
+            qtbot.keyPress(prompt._lineedit, key)
+        prompt._set_fileview_root(prompt._lineedit.text())
+
+        num_rows = prompt._file_model.rowCount(prompt._file_view.rootIndex())
+        visible = []
+        for row in range(num_rows):
+            parent = prompt._file_model.index(
+                os.path.dirname(prompt._lineedit.text()))
+            index = prompt._file_model.index(row, 0, parent)
+            if not prompt._file_view.isRowHidden(index.row(), index.parent()) and index.data() != "..":
```

Comment:
To make the test a bit simpler (as well as more strict), I'd prefer adding `".."` to the `expected` argument (after all, we expect it to be visible!) rather than excluding it here.

Reply:
Done.